tensorboard/webapp/routes/dashboard_deeplink_provider.ts

```diff
@@ -148,13 +137,35 @@ export class DashboardDeepLinkProvider extends DeepLinkProvider {
           return [{key: RUN_FILTER_KEY, value}];
         })
       ),
-      store
-        .select(selectors.getUnknownQueryParams)
-        .pipe(
-          map((queryParams) =>
-            Object.entries(queryParams).map(([key, value]) => ({key, value}))
-          )
-        ),
+      combineLatest([
+        store.select(getOverriddenFeatureFlags),
+        store.select(getFeatureFlagsMetadata),
+      ]).pipe(
+        map(([overriddenFeatureFlags, featureFlagsMetadata]) => {
+          return featureFlagsToSerializableQueryParams(
+            overriddenFeatureFlags,
+            featureFlagsMetadata
+          );
+        })
+      ),
+      store.select(selectors.getUnknownQueryParams).pipe(
+        withLatestFrom(store.select(getFeatureFlagsMetadata)),
+        map(([queryParams, featureFlags]) => {
+          const featureFlagKeys = new Set(
+            Object.values(featureFlags).map(
+              (featureFlag) => featureFlag.queryParamOverride
+            )
+          );
+          return (
+            Object.entries(queryParams)
+              // deserializeQueryParams doesn't check feature flag metadata before
```

Comment:
I don't have the context here, but isn't this the underlying issue instead? Can we make it so "non-unknown" parameters are not recognized as "unknown"? Or perhaps, does this come from the internal/external code separation?

Reply:
To Adrian's point, can we read the feature flag keys from `FeatureFlagMetadataMap` and ignore them in `deserializeQueryParams`?

Another idea: WDYT of filtering out all known keys (not just feature flags) from the unknown flag list (something like [this](https://github.com/tensorflow/tensorboard/compare/master...hoonji:tensorboard:filter_all_known_params?expand=1)) as a guard against similar bugs in the future? (And maybe log a warning that this is happening because it's probably an indication that `deserializeQueryParams` needs to be updated)

Reply:
Right, I hadn't anticipated these refactoring hurdles. I agree should focus on fixing the bug first, which seems to be making `showFlags=` unusable in local dev.

Thanks for creating the fix!